python-client/giskard/ml_worker/testing/registry/giskard_test.py

```diff
@@ -109,14 +109,18 @@ def __init__(self, test_fn: Function) -> None:
             meta = tests_registry.get_test(test_uuid)
         super(GiskardTest, self).__init__(test_fn, meta)
 
-    def __call__(self, *args, **kwargs) -> 'GiskardTestMethod':
-        self.is_initialized = True
-        self.params = kwargs
+    def __call__(self, *args, **kwargs) -> Result:
+        return self.test_fn(*args, **kwargs)
+
+    def set_params(self, *args, **kwargs) -> 'GiskardTestMethod':
```

Comment:
Actually it's to avoid something like this:

```
test_1 = test_f1.set_param(threshold=0.8)
test_2 = test_f1.set_param(threshold=0.5)

print(test_1.params['threshold']) => 0.5
```

Reply:
For me we should keep `GiskardTestMethod` as immutable for this reason. Maybe we can rename `set_params` to `with_params` or `fixture` to make it less confusing.